### DISCUSSION TITLE
V9.0.4 Time-Series Left/Right Y-Axis Align Zero (0)

### DISCUSSION THREAD

**Discussion by freebsdfrau** [grafana/grafana#54342] [Dashboards, Panels & Visualizations]
Grafana version 9.0.4 (c256012)

Hello, I recently needed to display some time-series data on a single graph showing both command count (a high number) and command queue length (a small number) and the best solution ended up being left/right Y-axes, however they are not aligning. Sometimes they align if I change the time range and refresh enough, but more than 90% of the time I would estimate, they do not align. Sometimes it's not a problem, but for colleagues that look at a graph like this and don't know to "look for 0 on the right versus left axis label" to confirm mis-aligned zeroes, then they might misinterpret the graph. This has led to all sorts of design decisions that I consider to be unnecessary if the axes would just align their zeroes. For example, having to use a zero fill on bars with non-zero line width to make hollow bars for more easily identifying misaligned axes, increasing the fill opacity of the bezier curves, etc.

![Screen Shot 2022-08-26 at 10 05 20 AM](https://user-images.githubusercontent.com/14223049/186957409-14734d3a-a44f-4693-8085-4576a9b690e3.png)

Same image with callouts highlighting the unaligned axes:

![Screen Shot 2022-08-26 at 10 05 20 AM copy](https://user-images.githubusercontent.com/14223049/186957968-19b96c4d-ee4c-474b-8b0e-3fb23360aa9f.png)

If I change the time range from last 24h to last 6h, we see it can be all over the place):

![Screen Shot 2022-08-26 at 10 18 09 AM](https://user-images.githubusercontent.com/14223049/186958340-785850ea-15ac-4044-8984-2932d633630e.png)

But then, if I zoom all the way out to last 2 days, it actually starts to look how I want it to look:

![Screen Shot 2022-08-26 at 10 21 50 AM](https://user-images.githubusercontent.com/14223049/186958813-1614d503-f78f-4a85-8afd-228a2030d057.png)

And then zooming back out to last 7 days, we can see again that things go awry:

![Screen Shot 2022-08-26 at 10 23 31 AM](https://user-images.githubusercontent.com/14223049/186959181-a9213e18-5ea5-498e-9b6f-ceb28663af05.png)

Is there some way to fix this?

**Comment by freebsdfrau** [comment]
This was solved by upgrading to Grafana 9.1.1 (released to FreeBSD ports less than 24h ago).

**Reply by freebsdfrau** [reply]
I spoke too soon -- there appears to be a bug and it doesn't always work

**Reply by ZhengfengRao** [reply]
how about this issue now? i'm facing the same problem

**Reply by ZhengfengRao** [reply]
t tried the lasted version, it works perfect, thank you

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
